In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

**Exploración inicial**

In [4]:
df_INAH_visitas = pd.read_csv('../data/raw/INAH_visitantes_museos_general.csv')
pd.set_option('display.max_columns', None)
df_INAH_visitas.head(5)

,anio,estado,clave_siinah,recinto,enero_nac,enero_ext,febrero_nac,febrero_ext,marzo_nac,marzo_ext,abril_nac,abril_ext,mayo_nac,mayo_ext,junio_nac,junio_ext,julio_nac,julio_ext,agosto_nac,agosto_ext,septiembre_nac,septiembre_ext,octubre_nac,octubre_ext,noviembre_nac,noviembre_ext,diciembre_nac,diciembre_ext
0,2025,Aguascalientes,2300100,Museo Regional de Historia de Aguascalientes,967,33,1721,27,1658,29,3726,61,2076,21,1378,50,1900,76,1571,46,1643,41,2959,66,3349,61,1660,61
1,2025,Baja California,2400100,"Museo Histórico Regional, Ex Cuartel de la Com...",1304,410,997,414,1932,292,1755,186,2544,409,3240,386,2100,511,2852,480,1649,132,3774,478,1784,516,1159,832
2,2025,Baja California Sur,2300200,Museo Regional de Antropología e Historia de B...,2266,983,1868,1101,2635,806,2337,544,2004,348,2009,547,2188,476,1040,567,940,346,2090,491,1969,515,1214,737
3,2025,Baja California Sur,2400200,Museo de Las Misiones Jesuíticas,659,179,887,474,798,551,1503,431,547,42,675,43,828,17,547,23,324,14,589,325,712,302,887,470
4,2025,Baja California Sur,2400300,Museo de las Pinturas Rupestres de San Ignacio,36,0,73,4,0,0,356,0,0,0,10,0,177,93,27,0,0,0,25,0,130,18,119,0


In [5]:
df_INAH_visitas.info()

<class 'pandas.DataFrame'>
RangeIndex: 3008 entries, 0 to 3007
Data columns (total 28 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   anio            3008 non-null   int64
 1   estado          3008 non-null   str  
 2   clave_siinah    3008 non-null   int64
 3   recinto         3008 non-null   str  
 4   enero_nac       3008 non-null   int64
 5   enero_ext       3008 non-null   int64
 6   febrero_nac     3008 non-null   int64
 7   febrero_ext     3008 non-null   int64
 8   marzo_nac       3008 non-null   int64
 9   marzo_ext       3008 non-null   int64
 10  abril_nac       3008 non-null   int64
 11  abril_ext       3008 non-null   int64
 12  mayo_nac        3008 non-null   int64
 13  mayo_ext        3008 non-null   int64
 14  junio_nac       3008 non-null   int64
 15  junio_ext       3008 non-null   int64
 16  julio_nac       3008 non-null   int64
 17  julio_ext       3008 non-null   int64
 18  agosto_nac      3008 non-null   int64
 

El conjunto de datos cuenta con 3008 registros/filas y 28 columnas(2 str y 26 int).

In [6]:
cantidad_nulos = df_INAH_visitas.isnull().sum()
print(f'Cantidad de valores nulos: {cantidad_nulos}')

Cantidad de valores nulos: anio              0
estado            0
clave_siinah      0
recinto           0
enero_nac         0
enero_ext         0
febrero_nac       0
febrero_ext       0
marzo_nac         0
marzo_ext         0
abril_nac         0
abril_ext         0
mayo_nac          0
mayo_ext          0
junio_nac         0
junio_ext         0
julio_nac         0
julio_ext         0
agosto_nac        0
agosto_ext        0
septiembre_nac    0
septiembre_ext    0
octubre_nac       0
octubre_ext       0
noviembre_nac     0
noviembre_ext     0
diciembre_nac     0
diciembre_ext     0
dtype: int64


In [10]:
cantidad_duplicados = df_INAH_visitas.duplicated().sum()
print(f'Cantidad de registros/filas duplicadas: \n{cantidad_duplicados}')

Cantidad de registros/filas duplicadas: 
0


No se identificaron valores nulos y duplicados.

In [13]:
annios = df_INAH_visitas['anio'].value_counts().sort_index(ascending=False)
print(annios)

anio
2025    104
2024    103
2023    101
2022    101
2021    101
2020    101
2019    101
2018    101
2017    101
2016     99
2015    102
2014    100
2013    100
2012    100
2011     99
2010     98
2009     99
2008    100
2007    101
2006     97
2005     98
2004     99
2003     99
2002    101
2001     95
2000     95
1999    103
1998    103
1997    103
1996    103
Name: count, dtype: int64


In [14]:
estados = df_INAH_visitas['estado'].value_counts().sort_index(ascending=True)
print(estados)

estado
Aguascalientes          30
Baja California         30
Baja California Sur    118
Campeche               120
Chiapas                120
Chihuahua               30
Ciudad de México       270
Coahuila                30
Colima                  30
Estado de México       236
Guanajuato             230
Guerrero                90
Hidalgo                136
Jalisco                120
Michoacán              150
Morelos                121
Nayarit                 60
Nuevo León              30
Oaxaca                 214
Puebla                 345
Querétaro               30
Quintana Roo            25
San Luis Potosí         30
Sinaloa                 28
Sonora                  30
Tabasco                 30
Tamaulipas              30
Tlaxcala                30
Veracruz               172
Yucatán                 63
Zacatecas               30
Name: count, dtype: int64
